test/smoke/src/application.ts

```diff
@@ -61,20 +61,29 @@ export class Application {
 	}
 
 	async start(): Promise<any> {
+		this.options.workspacePath = this.options.folderPath;
 		await this._start();
 		await this.code.waitForElement('.explorer-folders-view');
 		await this.code.waitForActiveElement(`.editor-instance[id="workbench.editor.walkThroughPart"] > div > div[tabIndex="0"]`);
 	}
 
-	async restart(options: { workspaceOrFolder?: string, extraArgs?: string[] }): Promise<any> {
+	async restart(options: { workspaceFilePath?: string, userDataDir?: string, extraArgs?: string[] }): Promise<any> {
 		await this.stop();
 		await new Promise(c => setTimeout(c, 1000));
-		await this._start(options.workspaceOrFolder, options.extraArgs);
+		Object.keys(options).forEach(key => {
+			if (options[key] === null || options[key] === undefined) {
+				delete options[key];
+			}
+		});
+		this.options = { ...this.options, ...options };
```

Comment:
From @joaomoreno 

@sandy081 This mutation is not good. If the tests ever run in a different order, there will be different behaviour, because this is remembering old options.

Reply:
@joaomoreno I think options has to be updated otherwise they are outdated and tests might be accessing invalid options

Reply:
For example `workspacePath`, `workspaceFilePath` and others are subjected to change.

Reply:
May be options needs to be reset for each test?